traitsui/editors/api.py

```diff
@@ -7,11 +7,15 @@
 try:
     from .array_editor import ArrayEditor
 except ImportError as e:
-    if 'numpy' not in e.message:
```

Comment:
I think I slightly prefer the approach of "try and import the editor, and if it fails, then see if we can work out if it's because of numpy being missing"